Doc/library/sqlite3.rst

```diff
@@ -1425,33 +1440,41 @@ This section shows recipes for common adapters and converters.
 Controlling Transactions
 ------------------------
 
-The underlying ``sqlite3`` library operates in ``autocommit`` mode by default,
-but the Python :mod:`sqlite3` module by default does not.
-
-``autocommit`` mode means that statements that modify the database take effect
-immediately.  A ``BEGIN`` or ``SAVEPOINT`` statement disables ``autocommit``
-mode, and a ``COMMIT``, a ``ROLLBACK``, or a ``RELEASE`` that ends the
-outermost transaction, turns ``autocommit`` mode back on.
-
-The Python :mod:`sqlite3` module by default issues a ``BEGIN`` statement
-implicitly before a Data Modification Language (DML) statement (i.e.
-``INSERT``/``UPDATE``/``DELETE``/``REPLACE``).
-
-You can control which kind of ``BEGIN`` statements :mod:`sqlite3` implicitly
-executes via the *isolation_level* parameter to the :func:`connect`
-call, or via the :attr:`isolation_level` property of connections.
-If you specify no *isolation_level*, a plain ``BEGIN`` is used, which is
-equivalent to specifying ``DEFERRED``.  Other possible values are ``IMMEDIATE``
-and ``EXCLUSIVE``.
-
-You can disable the :mod:`sqlite3` module's implicit transaction management by
-setting :attr:`isolation_level` to ``None``.  This will leave the underlying
-``sqlite3`` library operating in ``autocommit`` mode.  You can then completely
-control the transaction state by explicitly issuing ``BEGIN``, ``ROLLBACK``,
-``SAVEPOINT``, and ``RELEASE`` statements in your code.
-
-Note that :meth:`~Cursor.executescript` disregards
-:attr:`isolation_level`; any transaction control must be added explicitly.
+The ``sqlite3`` module does not adhere to the transaction handling recommended
+by PEP 249.
+If the connection attribute :attr:`isolation_level` is not :const:`None`,
+the following implicit transaction handling is performed:
+New transactions are implicitly opened before
+:meth:`~Cursor.execute` and :meth:`~Cursor.executemany` executes any of the
+following statements:
+
+* INSERT
+* UPDATE
+* DELETE
+* REPLACE
+
+In addition, any pending transaction is implicitly committed in
+:meth:`~Cursor.executescript`, before execution of the given SQL script.
```

Comment:
```suggestion
The :meth:`~Cursor.executescript` method implicitly commits
any pending transaction before execution of the given SQL script.
```

Avoid potentially confusing "additionally", since the statements here are parallel to, not additive to the statements about the other two methods in the previous paragraph. Also, state the method it applies to out front so the distinction between the two paras is clearer, and this construction is a bit shorter, simpler, more direct and easier to understand, and avoids an extra comma.

Reply:
This is a good catch. The "in addition" would definitely add confusion.